qst/src/main/java/io/deephaven/qst/type/Type.java

```diff
@@ -12,19 +13,29 @@
 public interface Type<T> {
 
     /**
-     * Finds the {@link #knownTypes() known type}, or else creates a {@link CustomType custom type}.
+     * Finds the {@link #knownTypes() known type}, or else creates the relevant
+     * {@link NativeArrayType native array type} or {@link CustomType custom type}.
      *
      * @param clazz the class
      * @param <T> the generic type of {@code clazz}
      * @return the type
      */
     static <T> Type<T> find(Class<T> clazz) {
```

Comment:
Should we break if someone uses this with `DbArray.class`?
Are we happy that this gives weaker typing for native arrays than making the native array type directly?

Reply:
Yes - I've added a check against creating Db array custom types.

It's a good question if the user already knows they have an array type, but aren't sure which one. I added `NativeArrayType.find(Class<?>)`.